src/dev/encoder.cpp

```diff
@@ -0,0 +1,152 @@
+#include <EVT/dev/encoder.hpp>
```

Comment:
It seems like this class has more functions than it really needs. The idea of breaking down your code does make sense, but there are so many short functions that it is difficult to follow the logic flow